### PR TITLE
feat: List Comprehension: Implement all(), any(), none(), and single()

### PR DESCRIPTION
**Pull request by dehowef**
* functions will return t or f based on whether certain conditions are met.

* all() will return true if and only if all elements in the list match the WHERE condition

* any() will return true if and only if atleast one element in the list matches the WHERE condition

* none() will return true if and only if zero elements in the list match the WHERE condition

* single() will return true if and only if one element in the list matches the WHERE condition

### CODE REVIEW

**Comment by htlim** [comment]
Seems good to me. but, please add regression test.

**Comment by dehowef** [comment]
@htlim I have now added regression tests.

File: src/test/regress/expected/cypher_expr.out [hunk 1]
```diff
@@ -496,6 +496,127 @@ RETURN [x IN [0, 1, 2, 3, 4] WHERE x % 2 = 0 | x + 1];
  [1, 3, 5]
 (1 row)
 
+--List Predicate Function
+RETURN ALL(x in [] WHERE x = 0);
```
Comment (htlim) [inline src/test/regress/expected/cypher_expr.out]: @dehowef I wonder if this is the correct behavior.
I confirmed that Neo4j works the same. However, it does not make sense that the 'ALL' operation on an empty list is true.

Comment (dehowef) [reply src/test/regress/expected/cypher_expr.out]: I would argue that the statement is vacuously true. On an empty set, all cases fulfill the requirements for ALL and NONE statements.

**Pull request closed without merging**